src/realm/object-store/sync/sync_session.cpp

```diff
@@ -109,6 +109,13 @@ void SyncSession::become_active()
     // when entering from the Dying state the session will still be bound
     if (!m_session) {
         create_sync_session();
+        // If the session failed to be created (e.g. location was not successfully updated),
+        // then revert back to the inactive state - the error was already passed to the provided
```

Comment:
the error you mention here, is it the one received if the location was not successfully updated?

Reply:
I moved the check for a valid location to the `handle_refresh()` function that returns the callback used for processing the update access token response. This has the actual failure that occurred that can be passed back to the user. And `create_sync_session()` now just has a `REALM_ASSERT()` that verifies the sync_route is not empty.